packages/components/src/custom-select-control-v2/types.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import type * as Ariakit from '@ariakit/react';
+
+export type CustomSelectContext =
+	| {
+			/**
+			 * The store object returned by Ariakit's `useSelectStore` hook.
+			 */
+			store: Ariakit.SelectStore;
+	  }
+	| undefined;
+
+export type CustomSelectProps = {
+	/**
+	 * The child elements. This should be composed of CustomSelect.Item components.
+	 */
+	children: React.ReactNode;
+	/**
+	 * An optional default value for the control. If left `undefined`, the first
+	 * non-disabled item will be used.
+	 */
+	defaultValue?: string | string[];
+	/**
+	 * Label for the control.
+	 */
+	label?: string;
+	/**
+	 * A function that receives the new value of the input.
+	 */
+	onChange?: ( newValue: string | string[] ) => void;
+	/**
+	 * Can be used to render select UI with custom styled values.
+	 */
+	styledValue?: ( selectValue: string | string[] ) => React.ReactNode;
```

Comment:
This prop name could be misleading, since it could give the impression that it affects all places where a `value` is rendered, while it's actually only used to render UI for the currently selected value.

A better name, IMO, would be something like `renderSelectedValue`. The "render" part is useful to hint that this is a render prop

Reply:
Sounds good! I'll change it back to `renderSelectedValue`, which was the name you originally used when we paired.

Reply:
Haha well I had forgotten about choosing the same exact name 😅  Anyway, naming things is very hard and I'm always open to suggestions for a better prop name!